rfc/0014-attribute-based-valuetypes/README.md

```diff
@@ -0,0 +1,88 @@
+<!--
+SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+
+SPDX-License-Identifier: AGPL-3.0-only
+-->
+
+# RFC 0014: Attributed-based syntax for defining value types
+
+| | |
+|---|---|
+| Feature Tag | `attribute-based value types` | <!-- TODO: choose a unique and declarative feature name -->
+| Status | `DRAFT` | <!-- Possible values: DRAFT, DISCUSSION, ACCEPTED, REJECTED -->
+| Responsible | `dirkriehle` | <!-- TODO: assign yourself as main driver of this RFC -->
+<!-- 
+  Status Overview:
+  - DRAFT: The RFC is not ready for a review and currently under change. Feel free to already ask for feedback on the structure and contents at this stage.
+  - DISCUSSION: The RFC is open for discussion. Usually, we open a PR to trigger discussions.
+  - ACCEPTED: The RFC was accepted. Create issues to prepare implementation of the RFC.
+  - REJECTED: The RFC was rejected. If another revision emerges, switch to status DRAFT.
+-->
+
+## Summary
+
+I'd like to allow the definition of user-defined single-attribute value types. In addition to the current syntax, in which an underlying value type is referenced through 'oftype', it does not use instantiation/inheritance but rather composition. 
+
+## Motivation
+
+The purpose of using composition is that it is
+
+1. more similar to traditional approaches and
+2. is needed anyway for multi-attribute value types.
+
+## Explanation
+
+To define a value type like CorrelationCoefficient and its range of -1 to +1, we have to write:
+
+```jayvee
+valuetype CorrelationCoefficient oftype decimal {
+  constraints: [
+    MinusOneToPlusOneRange,
+  ];
+}
+```
+
+This syntax is smart in that you don't have to list and name an attribute but rather rely on an implicit 'value' attribute. Still, I propose to make that attribute explicit. New syntax would be:
+
+```jayvee
+valuetype CorrelationCoefficient {
+  attributes: [
+    value oftype decimal;
+  ];
+  constraints: [
+    MinusOneToPlusOneRange; // Don't know how to attach this to value attribute
+  ];
+}
+```
+
+While more verbose, it prepares the way for 
+
+```jayvee
+valuetype Money {
+  attributes: [
+    amount oftype decimal;
+    currency oftype Currency;
+  ];
```

Comment:
We've used `property` instead of attributes in built-in blocks (https://github.com/jvalue/jayvee/tree/main/rfc/0008-builtin-blocktypes). I'd use the same naming here. For blocks, we decided properties are values an object owns intrinsically that you use in modeling, attributes is something you attribute to objects from the outside. The distinction is very faint but I would argue for consistency and reuse `property` here.

Otherwise I think the syntax is nice and would go for:

```
valuetype Money {
  property amount oftype decimal;
  property currency oftype Currency;
}
```

Reply:
At first glance, I'm indifferent/don't see a difference between property and attribute. Is there a semantic difference?

Which ever keyword we use, I observe a pattern of singular plural i.e. property/properties and constraint/constraints. 

My take is that users may want both (property for one new property, and properties for a list of properties, constraint for one constraint, and constraints for a set of constraints). If only one is possible, I'd go for singular i.e. Philip's suggestion